src/bun.js/bindings/napi.cpp

```diff
@@ -2738,17 +2739,17 @@ extern "C" napi_status napi_instanceof(napi_env env, napi_value object, napi_val
     return napi_set_last_error(env, napi_ok);
 }
 
-extern "C" napi_status napi_call_function(napi_env env, napi_value recv_napi,
-    napi_value func_napi, size_t argc,
+extern "C" napi_status napi_call_function(napi_env env, napi_value recv,
+    napi_value func, size_t argc,
     const napi_value* argv,
-    napi_value* result_ptr)
+    napi_value* result)
 {
     NAPI_PREAMBLE(env);
     NAPI_RETURN_EARLY_IF_FALSE(env, argc == 0 || argv, napi_invalid_arg);
-    JSValue funcValue = toJS(func_napi);
-    NAPI_RETURN_EARLY_IF_FALSE(env, funcValue.isObject(), napi_function_expected);
-    JSC::CallData callData = getCallData(funcValue);
-    NAPI_RETURN_EARLY_IF_FALSE(env, callData.type != JSC::CallData::Type::None, napi_function_expected);
+    JSValue funcValue = toJS(func);
+    if (!funcValue.isCallable() && !jsDynamicCast<AsyncContextFrame*>(funcValue)) {
+        return napi_invalid_arg;
+    }
```

Comment:
```suggestion
    NAPI_CHECK_ARG(env, func);
    JSValue funcValue = toJS(func);
    NAPI_RETURN_EARLY_IF_FALSE(env, funcValue.isCallable() || jsDynamicCast<AsyncContextFrame*>(funcValue), napi_invalid_arg);
```

Besides cleanup, the reasons to do this are:
- making the error code known to `napi_get_last_error_info`
- avoiding passing `nullptr` into `jsDynamicCast` (it will crash)

Also, in the original the error code was `napi_function_expected` not `napi_invalid_arg` -- was that wrong?

Reply:
nice, updated.

> Also, in the original the error code was napi_function_expected not napi_invalid_arg -- was that wrong?

yeah, node uses `CHECK_TO_FUNCTION` which surprisingly returns `napi_invalid_arg` instead of `napi_function_expected` when the input isn't a function.

Reply:
wow nice catch